tests/c/outline_recursion_indirect.newcg.c

```diff
@@ -0,0 +1,82 @@
+// ignore-if: test $YK_JIT_COMPILER != "yk" -o "$YKB_TRACER" = "swt"
+// Run-time:
+//   env-var: YKD_LOG_IR=-:aot,jit-pre-opt
+//   env-var: YKD_SERIALISE_COMPILATION=1
+//   env-var: YKD_LOG_JITSTATE=-
+//   stderr:
+//     jitstate: start-tracing
+//     0
+//     1
+//     2
+//     3
+//     4
+//     jitstate: stop-tracing
+//     --- Begin aot ---
+//     ...
+//     func main($arg0: i32, $arg1: ptr) -> i32 {
+//     ...
+//     --- End aot ---
+//     --- Begin jit-pre-opt ---
+//     ...
+//     Call @foo(%{{4}})
+//     ...
+//     --- End jit-pre-opt ---
+//     0
+//     1
+//     2
+//     3
+//     jitstate: enter-jit-code
+//     0
+//     1
+//     2
+//     0
+//     1
+//     jitstate: deoptimise
+//     0
+//     exit
+
+// Test outlining of recursive calls.
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <yk.h>
+#include <yk_testing.h>
+
+void bar(int i);
+
+void foo(int i) {
```

Comment:
Add the noinline attributes to the functions to ensure that it's indirect recursion.

Reply:
That was in the other test :)

Reply:
sorry, this was github rendering it wrong.